src/ElfReader.cc

```diff
@@ -317,7 +380,31 @@ ScopedFd ElfFileReader::open_debug_file(const std::string& elf_file_name) {
   string debug_path = "/usr/lib/debug/";
   debug_path +=
     elf_file_name.substr(0, last_slash) + '/' + debuglink.filename.data();
-  return ScopedFd(debug_path.c_str(), O_RDONLY);
+  ScopedFd debug_fd(debug_path.c_str(), O_RDONLY);
+  if (!debug_fd.is_open()) {
+    return ScopedFd();
+  }
+
+  uint32_t crc = 0xffffffff;
+  while (true) {
+    unsigned char buf[4096];
+    ssize_t ret = ::read(debug_fd.get(), buf, sizeof(buf));
+    if (ret < 0) {
+      if (errno != EINTR) {
+        LOG(debug) << "Error reading " << debug_path;
+        return ScopedFd();
+      }
+    } else if (ret == 0) {
+      break;
+    } else {
+      crc = crc32(crc, buf, ret);
+    }
+  }
+
+  if ((crc ^ 0xffffffff) == debuglink.crc) {
+    return debug_fd;
+  }
+  return ScopedFd();
```

Comment:
I think we should probably go ahead and do the CRC check but we should only do the debuginfo search if we failed to find `.symtab` in the main executable. Please add a comment explaining why we're doing the CRC check.